dGame/dComponents/PropertyEntranceComponent.cpp

```diff
@@ -74,261 +67,222 @@ void PropertyEntranceComponent::OnEnterProperty(Entity* entity, uint32_t index,
-	property.IsFriend = isFriend;
-	property.IsModeratorApproved = isModeratorApproved;
-	property.IsAlt = isAlt;
-	property.IsOwned = isOwned;
-	property.AccessType = privacyOption;
-	property.DateLastPublished = timeLastUpdated;
-	property.PerformanceCost = performanceCost;
-
-	return property;
+void PropertyEntranceComponent::PopulateUserFriendMap(uint32_t user) {
+	this->m_UserFriendMap[user] = {};
+
+	auto friendQuery = Database::CreatePreppedStmt("SELECT * FROM friends WHERE friend_id = ? OR player_id = ?;");
+	friendQuery->setUInt(1, user);
+	friendQuery->setUInt(2, user);
+
+	auto friendRes = friendQuery->executeQuery();
+
+	while (friendRes->next()) {
+		auto friendId = friendRes->getUInt(2);
+		auto playerId = friendRes->getUInt(1);
+		auto bestFriend = friendRes->getUInt(3) == 3;
+
+		if (friendId == user) {
+			this->m_UserFriendMap[user][playerId] = bestFriend;
+		} else {
+			this->m_UserFriendMap[user][friendId] = bestFriend;
+		}
+	}
+
+	delete friendRes;
+	friendRes = nullptr;
+
+	delete friendQuery;
+	friendQuery = nullptr;
 }
 
-std::string PropertyEntranceComponent::BuildQuery(Entity* entity, int32_t sortMethod, Character* character, std::string customQuery, bool wantLimits) {
-	std::string base;
-	if (customQuery == "") {
-		base = baseQueryForProperties;
+std::vector<uint32_t> PropertyEntranceComponent::GetPropertyIDsBasedOnParams(const std::string& searchText, uint32_t sortMethod, Entity* requestor) {
+	std::string query = "SELECT id, owner_id, last_updated, reputation FROM properties WHERE zone_id = ? AND (description LIKE ? OR name LIKE ? OR name LIKE ?) AND privacy_option >= ? ";
+
+	if (sortMethod == SORT_TYPE_REPUTATION) {
+		query += "ORDER BY reputation DESC;";
 	} else {
-		base = customQuery;
+		query += "ORDER BY last_updated DESC;";
 	}
-	std::string orderBy = "";
-	if (sortMethod == SORT_TYPE_FEATURED || sortMethod == SORT_TYPE_FRIENDS) {
-		std::string friendsList = " AND p.owner_id IN (";
 
-		auto friendsListQuery = Database::CreatePreppedStmt("SELECT * FROM (SELECT CASE WHEN player_id = ? THEN friend_id WHEN friend_id = ? THEN player_id END AS requested_player FROM friends ) AS fr WHERE requested_player IS NOT NULL ORDER BY requested_player DESC;");
+	auto propertyLookup = Database::CreatePreppedStmt(query);
 
-		friendsListQuery->setUInt(1, character->GetID());
-		friendsListQuery->setUInt(2, character->GetID());
+	const auto searchString = "%" + searchText + "%";
+	propertyLookup->setUInt(1, this->m_MapID);
+	propertyLookup->setString(2, searchString.c_str());
+	propertyLookup->setString(3, searchString.c_str());
+	propertyLookup->setString(4, searchString.c_str());
+	propertyLookup->setInt(5, sortMethod == SORT_TYPE_FEATURED || sortMethod == SORT_TYPE_FRIENDS ? (uint32_t)PropertyPrivacyOption::Friends : (uint32_t)PropertyPrivacyOption::Public);
+
+	auto propertyIdRes = propertyLookup->executeQuery();
 
-		auto friendsListQueryResult = friendsListQuery->executeQuery();
+	std::vector<uint32_t> propertyIds{};
```

Comment:
nit: You dont need an initializer list constructor here, std::vector on the stack will already be default initialized